ctpolicy/ctpolicy.go

```diff
@@ -49,13 +51,102 @@ func (group *LogGroupInfo) setMinInclusions(i int) error {
 
 func (group *LogGroupInfo) populate(ll *loglist.LogList, included func(log *loglist.Log) bool) {
 	group.LogURLs = make(map[string]bool)
+	group.LogWeights = make(map[string]float32)
 	for _, l := range ll.Logs {
 		if included(&l) {
 			group.LogURLs[l.URL] = true
+			group.LogWeights[l.URL] = 1.0
 		}
 	}
 }
 
+// resetLogWeights applies suggested weights to the Log-group. Does not reset
+// weights and returns error when not enough to reach minimal inclusion number
+// positive weights provided.
+func (group *LogGroupInfo) ResetLogWeights(weights map[string]float32) error {
+	groupWeights := make(map[string]float32, len(group.LogURLs))
+	for logURL := range group.LogURLs {
+		groupWeights[logURL] = 1.0
+	}
+	nonZeroNum := len(group.LogURLs)
+	for logURL, w := range weights {
```

Comment:
This could also use the refactored satisfiability function

Reply:
Extracted.